js/models/messages.js

```diff
@@ -1935,105 +1977,212 @@
+        const shouldAutoAcceptFR = await this.handleSecondaryDeviceFriendRequest(
+          source
+        );
+        if (shouldAutoAcceptFR) {
+          libloki.api.debug.logAutoFriendRequest(
+            `Received AUTO_FRIEND_REQUEST from source: ${source}`
+          );
+          // Directly setting friend request status to skip the pending state
+          await conversation.setFriendRequestStatus(
+            window.friends.friendRequestStatusEnum.friends
+          );
+          // sending a message back = accepting friend request
+
+          window.libloki.api.sendBackgroundMessage(
+            source,
+            window.textsecure.OutgoingMessage.DebugMessageType.AUTO_FR_ACCEPT
+          );
+          confirm();
+          // return true to notify the message is fully processed
+          return true;
         }
       }
+      return false;
+    },
 
-      const isSessionRequest =
-        initialMessage.flags ===
-        textsecure.protobuf.DataMessage.Flags.SESSION_REQUEST;
+    async handleDataMessage(initialMessage, confirm) {
+      // This function is called from the background script in a few scenarios:
+      //   1. on an incoming message
+      //   2. on a sent message sync'd from another device
+      //   3. in rare cases, an incoming message can be retried, though it will
+      //      still go through one of the previous two codepaths
+      const ourNumber = textsecure.storage.user.getNumber();
+      const message = this;
+      const source = message.get('source');
+      let conversationId = message.get('conversationId');
+      const authorisation = await libloki.storage.getGrantAuthorisationForSecondaryPubKey(
+        source
+      );
+      const primarySource =
+        (authorisation && authorisation.primaryDevicePubKey) || source;
+      const isGroupMessage = !!initialMessage.group;
+      if (isGroupMessage) {
+        /* handle one part of the group logic here:
+           handle requesting info of a new group,
+           dropping an admin only update from a non admin, ...
+         */
+        conversationId = initialMessage.group.id;
+        const shouldReturn = this.handleGroupMessage(
+          source,
+          initialMessage,
+          primarySource,
+          confirm
+        );
+
+        // handleGroupMessage() can process fully a message in some cases
+        // so we need to return early if that's the case
+        if (shouldReturn) {
+          return null;
+        }
+      } else if (source !== ourNumber && authorisation) {
+        // Ignore auth from our devices
+        conversationId = authorisation.primaryDevicePubKey;
+      }
+
+      // the conversation with the primary device of that source (can be the same as conversationOrigin)
+      const conversationPrimary = ConversationController.get(conversationId);
+      // the conversation with this real device
+      const conversationOrigin = ConversationController.get(source);
```

Comment:
some calls in https://github.com/loki-project/session-desktop/blob/clearnet/libtextsecure/message_receiver.js#L1131 are on the conversation with the source device (not the primary one)